src/emulation/mod.rs

```diff
@@ -137,18 +139,47 @@ impl SyscallMock {
     }
 }
 
+pub enum Executable {
+    Interpreter(PathBuf),
+    Program(PathBuf),
+}
+
+impl Deref for Executable {
+    type Target = Path;
+    fn deref(&self) -> &Path {
+        match self {
+            Executable::Interpreter(program) => &program,
+            Executable::Program(program) => &program,
+        }
```

Comment:
We implement `Deref` on the `Executable` enum so that we can coerce it into a `&Path` when calling `execve`

Reply:
This comment is now out of date, found a better way to do this :)